AK/IpAddressCidr.h

```diff
@@ -135,8 +135,11 @@ class IPv4AddressCidr : public Details::IPAddressCidr<IPv4AddressCidr> {
         if (free_bits == 32) {
             netmask = IPv4Address(0, 0, 0, 0);
         } else {
-            u32 mask = -1;
-            netmask = IPv4Address(mask >> free_bits);
+            u8 address[4];
+            NetworkOrdered<u32> mask = NumericLimits<u32>::max() << free_bits;
+
+            memcpy(address, &mask, sizeof(address));
+            netmask = IPv4Address(address);
```

Comment:
The entire third commit should be split and merged into the IPv6/IPv4 commits respectively.